tests/core/input/test_input_event_key.h

```diff
@@ -115,81 +123,84 @@ TEST_CASE("[InputEventKey] Key correctly converts itself to text") {
 	// Set physical key to Escape.
-	none_key.set_physical_keycode(Key::ESCAPE);
-	CHECK(none_key.to_string() == "InputEventKey: keycode=4194305 (Escape), mods=none, physical=true, location=unspecified, pressed=false, echo=false");
+	none_key->set_physical_keycode(Key::ESCAPE);
+	CHECK(none_key->to_string() == "InputEventKey: keycode=4194305 (Escape), mods=none, physical=true, location=unspecified, pressed=false, echo=false");
 
-	InputEventKey key;
+	Ref<InputEventKey> key;
+	key.instantiate();
 
 	// Set physical to None, set keycode to Space.
-	key.set_keycode(Key::SPACE);
-	CHECK(key.to_string() == "InputEventKey: keycode=32 (Space), mods=none, physical=false, location=unspecified, pressed=false, echo=false");
+	key->set_keycode(Key::SPACE);
+	CHECK(key->to_string() == "InputEventKey: keycode=32 (Space), mods=none, physical=false, location=unspecified, pressed=false, echo=false");
 
 	// Set location
-	key.set_location(KeyLocation::RIGHT);
-	CHECK(key.to_string() == "InputEventKey: keycode=32 (Space), mods=none, physical=false, location=right, pressed=false, echo=false");
+	key->set_location(KeyLocation::RIGHT);
+	CHECK(key->to_string() == "InputEventKey: keycode=32 (Space), mods=none, physical=false, location=right, pressed=false, echo=false");
 
 	// Set pressed to true.
-	key.set_pressed(true);
-	CHECK(key.to_string() == "InputEventKey: keycode=32 (Space), mods=none, physical=false, location=right, pressed=true, echo=false");
+	key->set_pressed(true);
+	CHECK(key->to_string() == "InputEventKey: keycode=32 (Space), mods=none, physical=false, location=right, pressed=true, echo=false");
 
 	// set echo to true.
-	key.set_echo(true);
-	CHECK(key.to_string() == "InputEventKey: keycode=32 (Space), mods=none, physical=false, location=right, pressed=true, echo=true");
+	key->set_echo(true);
+	CHECK(key->to_string() == "InputEventKey: keycode=32 (Space), mods=none, physical=false, location=right, pressed=true, echo=true");
 
 	// Press Ctrl and Alt.
-	key.set_ctrl_pressed(true);
-	key.set_alt_pressed(true);
+	key->set_ctrl_pressed(true);
+	key->set_alt_pressed(true);
 #ifdef MACOS_ENABLED
-	CHECK(key.to_string() == "InputEventKey: keycode=32 (Space), mods=Ctrl+Option, physical=false, location=right, pressed=true, echo=true");
+	CHECK(key->to_string() == "InputEventKey: keycode=32 (Space), mods=Ctrl+Option, physical=false, location=right, pressed=true, echo=true");
 #else
-	CHECK(key.to_string() == "InputEventKey: keycode=32 (Space), mods=Ctrl+Alt, physical=false, location=right, pressed=true, echo=true");
+	CHECK(key->to_string() == "InputEventKey: keycode=32 (Space), mods=Ctrl+Alt, physical=false, location=right, pressed=true, echo=true");
 #endif
 }
 
 TEST_CASE("[InputEventKey] Key is correctly converted to reference") {
-	InputEventKey base_key;
-	Ref<InputEventKey> key_ref = base_key.create_reference(Key::ENTER);
+	Ref<InputEventKey> key_ref = InputEventKey::create_reference(Key::ENTER);
```

Comment:
This case is a bit complicated and seems to depend on possible misunderstanding of the way this method works, removed `base_key` as it is irrelevant here